pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -326,6 +331,14 @@ func (s *NodeController) checkNodeReady(node *api.Node) *api.NodeCondition {
 	}
 }
 
+// sendConditionEvent sends condition change event.
+func (s *NodeController) sendConditionEvent(node *api.Node, kind api.NodeConditionKind, oldCondition, newCondition *api.NodeCondition) {
+	// TODO: if 'oldCondition' is nil, we should publish the initial condition, but selfLink is not set.
+	if oldCondition != nil {
+		record.Eventf(node, "condition change", "Node condition %v changes from %v to %v", kind, oldCondition.Status, newCondition.Status)
```

Comment:
Gocha!  Thanks for the tip.  I've changed to "changedToFullReadyCondition", "changedToNoneReachableCondition", etc.  This should be descriptive.  Feel free to object if this doesn't look good.

Working on test.